backend/modules/pdf_processing/router.py

```
@@ -1 +1,287 @@
+            pipeline = get_pipeline(db, current_user)
+            await pipeline.process_full_with_progress(file, session_id, context_files)
+        except HTTPException as e:
+            debug_log(f"[ENDPOINT] HTTPException in background task: {e.status_code} - {e.detail}")
+            if session_id:
+                progress_manager.error_processing(session_id, f"{e.detail}")
+        except Exception as e:
+            debug_log(f"[ENDPOINT] Exception in background task: {e}")
+            if session_id:
+                progress_manager.error_processing(session_id, f"PDF parsing failed: {str(e)}")
+    
+    asyncio.create_task(run_parsing())
```

Ruff (0.14.5)

Store a reference to the return value of `asyncio.create_task`

(RUF006)